crash when trying `\copy` command

**Describe the problem**
I was trying to see if cockroachdb support `\copy` command

```
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1  a panic has occurred!
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +interface conversion: *pq.stmt is not clisqlclient.copyFromer: missing method CopyData
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +(1) attached stack trace
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  -- stack trace:
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | runtime.assertI2I
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	GOROOT/src/runtime/iface.go:458
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlclient.BeginCopyFrom
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlclient/copy.go:45
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).beginCopyFrom
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1784
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doHandleCliCmd.func1
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1231
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runWithInterruptableCtx
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2201
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doHandleCliCmd
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1230
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doRunShell
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1885
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runIncludeInternal
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1562
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runString
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1535
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runStatements
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2029
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doRunShell
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1856
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).RunInteractive
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1839
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clisqlcfg.(*Context).Run
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clisqlcfg/context.go:205
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.runTerm
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/sql_shell_cmd.go:63
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli/clierrorplus.MaybeDecorateError.func1
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:67
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/spf13/cobra.(*Command).execute
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/spf13/cobra.(*Command).ExecuteC
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/spf13/cobra.(*Command).Execute
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.Run
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:296
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.doMain
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:138
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.Main
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:65
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | main.main
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	main/pkg/cmd/cockroach/main.go:26
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | runtime.main
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	GOROOT/src/runtime/proc.go:255
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | runtime.goexit
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +  | 	GOROOT/src/runtime/asm_amd64.s:1581
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +Wraps: (2) interface conversion: *pq.stmt is not clisqlclient.copyFromer: missing method CopyData
E220530 23:19:49.992756 1 1@util/log/logcrash/crash_reporting.go:174  [-] 1 +Error types: (1) *withstack.withStack (2) *runtime.TypeAssertionError
panic: interface conversion: *pq.stmt is not clisqlclient.copyFromer: missing method CopyData [recovered]
	panic: interface conversion: *pq.stmt is not clisqlclient.copyFromer: missing method CopyData

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.RecoverAndReportPanic({0x62df9e8, 0xc00007c118}, 0xc001729a40)
	github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:124 +0x78
panic({0x477ff00, 0xc001606420})
	GOROOT/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/cli/clisqlclient.BeginCopyFrom({0x62df9e8, 0xc00007c118}, {0x639f390, 0xc0017cedd0}, {0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlclient/copy.go:45 +0x1b2
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).beginCopyFrom(0xc000ed6280, {0x62df9e8, 0xc00007c118}, {0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1784 +0x7f
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doHandleCliCmd.func1({0x62df9e8, 0xc00007c118})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1231 +0x3e
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runWithInterruptableCtx(0xc000a08cab, 0x5)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2201 +0x1df
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doHandleCliCmd(0xc000ed6280, 0x2, 0x128)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1230 +0xe88
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doRunShell(0xc000ed6280, 0x17295f8, 0x46b66d, 0x14, 0x0)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1885 +0x314
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runIncludeInternal(0xc000ed6140, 0x5, 0x4, {0x4e1de08, 0x2}, 0xc0014286c0, 0x0, 0x1)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1562 +0x1c5
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runString(0xc000ed6140, 0x0, 0xc001729878, {0x7ffecf3d60fa, 0x5})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1535 +0x1bd
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).runStatements(0xc000ed6140, {0xc00097cf80, 0x1, 0x99df0e0})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2029 +0x14b
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).doRunShell(0xc000ed6140, 0x60, 0x7f30bffc05b8, 0x60, 0xc000500400)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1856 +0x178
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).RunInteractive(0xc0017cedd0, 0x639f390, 0xc0017cedd0, 0x3f7cbef)
	github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:1839 +0x6d
github.com/cockroachdb/cockroach/pkg/cli/clisqlcfg.(*Context).Run(0xc000e14b60, {0x639f390, 0xc0017cedd0})
	github.com/cockroachdb/cockroach/pkg/cli/clisqlcfg/context.go:205 +0x21e
github.com/cockroachdb/cockroach/pkg/cli.runTerm(0xc001729ae0, {0x4fd4d1, 0x0, 0x500786})
	github.com/cockroachdb/cockroach/pkg/cli/sql_shell_cmd.go:63 +0x15d
github.com/cockroachdb/cockroach/pkg/cli/clierrorplus.MaybeDecorateError.func1(0x907f820, {0xc0018181e0, 0x0, 0x3})
	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:67 +0x43
github.com/spf13/cobra.(*Command).execute(0x907f820, {0xc0018181b0, 0x3, 0x3})
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x9071ea0)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
github.com/cockroachdb/cockroach/pkg/cli.Run(...)
	github.com/cockroachdb/cockroach/pkg/cli/cli.go:296
github.com/cockroachdb/cockroach/pkg/cli.doMain(0x907f820, {0xc0001204ea, 0x3})
	github.com/cockroachdb/cockroach/pkg/cli/cli.go:138 +0x1e9
github.com/cockroachdb/cockroach/pkg/cli.Main()
	github.com/cockroachdb/cockroach/pkg/cli/cli.go:65 +0x136
main.main()
	main/pkg/cmd/cockroach/main.go:26 +0x17
```

**To Reproduce**
```bash
cockroach sql --insecure -e '\copy'
```

**Expected behavior**
Not crash

**Additional data / screenshots**
```
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
# Server version: CockroachDB CCL v22.1.0 (x86_64-pc-linux-gnu, built 2022/05/23 16:27:47, go1.17.6) (same version as client)
# Cluster ID: a62f56e4-992c-4ce8-b1f4-d87f7d65a52f
No entry for terminal type "xterm-256color";
using dumb terminal settings.
#
# Enter \? for a brief introduction.
#
root@:26257/defaultdb> select version();
                                       version
--------------------------------------------------------------------------------------
  CockroachDB CCL v22.1.0 (x86_64-pc-linux-gnu, built 2022/05/23 16:27:47, go1.17.6)
(1 row)


Time: 29ms total (execution 23ms / network 5ms)

```


Jira issue: CRDB-16206

--- a/pkg/cli/clisqlshell/sql.go
+++ b/pkg/cli/clisqlshell/sql.go
@@ -1235,13 +1235,16 @@ func (c *cliState) doHandleCliCmd(loopState, nextState cliStateEnum) cliStateEnu
 
 	case `\copy`:
 		c.exitErr = c.runWithInterruptableCtx(func(ctx context.Context) error {
-			return c.beginCopyFrom(ctx, c.concatLines)
+			// Strip out the starting \ in \copy.
+			return c.beginCopyFrom(ctx, line[1:])
 		})
-		if !c.singleStatement {
-			clierror.OutputError(c.iCtx.stderr, c.exitErr, true /*showSeverity*/, false /*verbose*/)
-		}
-		if c.exitErr != nil && c.iCtx.errExit {
-			return cliStop
+		if c.exitErr != nil {
+			if !c.singleStatement {
+				clierror.OutputError(c.iCtx.stderr, c.exitErr, true /*showSeverity*/, false /*verbose*/)
+			}
+			if c.iCtx.errExit {
+				return cliStop
+			}
 		}
 		return cliStartLine
 

--- a/pkg/cli/interactive_tests/test_copy.tcl
+++ b/pkg/cli/interactive_tests/test_copy.tcl
@@ -60,7 +60,11 @@ eexpect "COPY 3"
 eexpect root@
 
 # Try \copy as well.
-send "\copy t FROM STDIN CSV;\r"
+send "\\copy t FROM STDIN CSV INVALID;\r"
+eexpect "syntax error"
+eexpect root@
+
+send "\\copy t FROM STDIN CSV;\r"
 eexpect ">>"
 send "4,epa! epa!\r"
 send_eof

